### ISSUE TITLE
v6.1.9 freezes completely

### ISSUE THREAD

**Issue by reteP-riS** [nextcloud/ios#3243]
### Steps to reproduce
1. on iPad with iOS 18.2 update Nextcloud app from v6.1.8 to v6.1.9
2. start the app
3. touching any folder or file or menu item does NOTHING - the app is stone dead
4. closing the app or rebooting the iPad doesn't help

### Expected behaviour
The app should react to touch gestures!

### Actual behaviour
It's stone dead!

### Logs

2024-12-17 12:48:22 [INFO] Start session with level 1 Nextcloud Hydrogen for iOS 6.1.9.4 © 2024
2024-12-17 12:48:22 [INFO] Account active USERNAME https://nextcloud.DOMAIN.net/
2024-12-17 12:48:22 [INFO] Scene will enter in foreground
2024-12-17 12:48:23 [INFO] Scene did become active
2024-12-17 12:48:23 [INFO] Initialize Auto upload with 0 uploads

### Environment data

**iOS version:** iOS 18.2
**Nextcloud iOS app version:** v6.1.19
**Server operating system:** Debian GNU/Linux 11.11 (bullseye)
**Web server:** Apache
**Database:** MariaDB 10.5.22
**PHP version:** 8.2.26
**Nextcloud server version:** Nextcloud Hub 9 (30.0.4)

**Comment by mpivchev** [comment]
Hi, thanks for the report.

Have you found anything specific on how to reproduce this?
We have received other reports for this, and it has happened for me as well, but at the moment we are not sure what is causing it.

**Comment by reteP-riS** [comment]
Unfortunately not. Not even any entries in the Nextcloud server's errorlog.

**Comment by i2h3** [comment]
I tried to reproduce this issue but failed. Most of the factors like versions are the same in my case.

* **Device**: Physical 11" iPad 2024 with M4
* **App**: 6.1.9.4 (App Store Release)
* **iOS**: 18.2
* **Nextcloud Server**: Nextcloud Hub 9 (30.0.4) (All-in-One Deployment)

The app remains responsive, be it my personal test server or our corporate deployment.

**Comment by johnfgoodman** [comment]
Is it possible it is related to auto upload? On my wife's iPhone 13 (iOS 18.1.1), if I completely closed the app (by swiping it up on the app switcher) and then reopened it right away it would work fine. However, if I completely closed the app and then took a picture and then re-opened the app, it would freeze as described above. I was able to reproduce a few times, but then it stopped freezing for some reason. I found this out because I have been troubleshooting the auto upload that does not seem to be working consistently on her phone.

**Comment by PackDUde** [comment]
On my iPhone 16 Pro I can reproduce it. Moving a few small jpg images from on Nextcloud folder to another will cause the app to lock up for up to 15 seconds or so pretty reliably.

**Comment by PackDude** [comment]
Issue seems to be fixed in 6.2.0.1. Nice job folks!

**Comment by loeffert** [comment]
Can also confirm, that it works again with latest release 6.2.0

**Comment by reteP-riS** [comment]
v6.2.0 still works ok for me and I use it a lot. Was the issue maybe reintroduced with v6.2.1?

**Comment by mpivchev** [comment]
> v6.2.0 still works ok for me and I use it a lot. Was the issue maybe reintroduced with v6.2.1?

Yes. We use semaphores to limit these access on reloading list of files. This was causing a deadlock.
